internal/pkg/cli/pipeline_delete.go

```diff
@@ -116,23 +121,27 @@ func (o *deletePipelineOpts) Ask() error {
 			return err
 		}
 	}
+
 	if o.name != "" {
-		if _, err := o.codepipeline.GetPipeline(o.name); err != nil {
-			return err
+		pipeline, err := getDeployedPipelineInfo(o.deployedPipelineLister, o.appName, o.name)
+		if err != nil {
+			return fmt.Errorf("validate pipeline name %s: %w", o.name, err)
 		}
+		o.targetPipeline = pipeline
 	} else {
-		pipelineName, err := askDeployedPipelineName(o.sel, o.appName, fmt.Sprintf(fmtPipelineDeletePrompt, color.HighlightUserInput(o.appName)))
+		pipeline, err := askDeployedPipelineName(o.sel, fmt.Sprintf(fmtPipelineDeletePrompt, color.HighlightUserInput(o.appName)), o.appName)
 		if err != nil {
 			return err
 		}
-		o.name = pipelineName
+		o.name = pipeline.Name
+		o.targetPipeline = pipeline
 	}
 
 	if o.skipConfirmation {
 		return nil
 	}
 	deleteConfirmed, err := o.prompt.Confirm(
-		fmt.Sprintf(pipelineDeleteConfirmPrompt, o.name, o.appName),
+		fmt.Sprintf(pipelineDeleteConfirmPrompt, o.targetPipeline.Name, o.appName),
```

Comment:
I see in the other commands we use `o.name`. Ideally they should be equal but is there any reason why we use `o.targetPipeline.Name` here?